src/segment-loader.js

```diff
@@ -2031,6 +2034,27 @@ export default class SegmentLoader extends videojs.EventTarget {
     // as we use the start of the segment to offset the best guess (playlist provided)
     // timestamp offset.
     this.updateSourceBufferTimestampOffset_(segmentInfo);
+
+    // throw away the isSyncRequest segment if
+    // it wouldn't have been the next segment we request.
+    if (segmentInfo.isSyncRequest) {
+      this.syncController_.saveSegmentTimingInfo({
+        segmentInfo,
+        shouldSaveTimelineMapping: this.loaderType_ === 'main'
+      });
+
+      const next = this.chooseNextRequest_();
+
+      // if the sync request isn't the segment that would be request next
+      // after taking into account it's data. do not append it.
+      if (next.mediaIndex !== segmentInfo.mediaIndex || next.partIndex !== segmentInfo.partIndex) {
+        this.logger_('sync segment was incorrect, not appending');
+        return;
+      }
```

Comment:
question: this was causing us issues because we got a sync request for a location that was different from where we wanted but previously we were appending the sync segment but this caused problems?

Reply:
We download an entire segment for sync purposes. The segment used to always be inaccurate but appended anyway costing us a lot of time and back/forward buffer space. Now we only append if the segment would have been the next segment we choose anyway. Otherwise we save the timing info and throw away the segment than choose the real next segment.

Reply:
Makes sense. Maybe one day we'll be able to use `fetch` and only download enough of that segment to get timing info rather than waiting for the entire segment.